device.go

```diff
@@ -200,6 +200,7 @@ func (device *BlockDevice) attach(h hypervisor, c *Container) error {
 	// rootfs and the predicted block device name needs to be provided to the agent.
 	index, err := c.pod.getAndSetPodBlockIndex()
 	if err != nil {
+		c.pod.decrementPodBlockIndex()
```

Comment:
You probably want to defer a function that will check for err, and do the decrement if `err != nil`

Reply:
done